bitcoin/src/pow.rs

```diff
@@ -959,7 +959,8 @@ impl std::error::Error for GetBlockHeaderError {}
 pub fn calculate_next_work_required<F>(
     consensus_params: Params,
     current_height: Height,
-    current_time: u32,
+    current_header: Header,
```

Comment:
I am not sure that all the applications `calculate_next_work_required` have access to a header, I should check what we do in Nakamoto and when we use it.

Otherwise looks fine

Reply:
If you look at the function's usage, all you need is `time`, `bits`, and `prev_blockhash`

But `prev_blockhash` is only needed if you use testnet.

Also, you need `current_header.target()` which can be calculated with the `bits` value alone.

Reply:
Height is not contained in the Header, so it needs to be input separately.

Also note that `new_block_timestamp` is also only needed when using testnet.